### PR TITLE
Fix: tx confirmation gas validations

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-2686
could also fix RNBW-3208 @skylarbarrera do you have a path to repro?

## What changed (plus any additional context for devs)

This bug was happening because we have wrong logic making the gas validations.
To validate gas we take a native asset corresponding to the network the transaction is in, taking the `txNetwork` information from state, and using the account address coming from `useAccountSettings` from state, which is the current wallet the user is in. All good if you're transacting with the same wallet but in the TransactionConfirmationScreen we could have a tx for an account address different to the one in `useAccountSettings`, coming from the wallet connect session, in which case we would be validating against the wrong native asset balance, because is another wallet's balance.
 
To fix this i'm moving the gas validation to the `useGas` hook instead of having it in state, so we can pass a different `nativeasset` to validate gas with, used in `TransactionConfirmationScreen`. For all other places we continue using `ethereumUtils.getNetworkNativeAsset(network)` to get the asset for the wallet depending on the network.

## PoW (screenshots / screen recordings)

In the video you can see I'm in a wallet with 0 ETH but trying to sign tx with estebanmino.eth which has ETH, so I can send the tx. When I switch the wallet to  estebanmino.eth and connect with a wallet with 0 ETH, i'm not able to send the tx since i don't have ETH.

(don't take in account my ens PFP since I was working on profiles then jumped to this and it was already in state)

https://user-images.githubusercontent.com/12115171/167034160-27dbfd3d-5c06-48d9-9a9a-f3d34410399d.mp4

## Final checklist

- [x] Assigned individual reviewers?
- [ ] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

**Review by brunobar79 — COMMENTED** [review]
Looks good to me! 
Before moving fwd,  I'd like to get your input in the code reusage suggestions I mentioned.

File: src/hooks/useGas.ts [hunk 1]
```diff
@@ -12,8 +23,38 @@ import {
...
   gasUpdateTxFee,
 } from '@rainbow-me/redux/gas';
+import { ethereumUtils } from '@rainbow-me/utils';
+
+const checkSufficientGas = (
+  txFee: LegacyGasFee | GasFee,
+  network: Network,
+  nativeAsset?: any
+) => {
+  const isL2 = isL2Network(network);
```
[the remark this answers is not in the document]
Comment (estebanmino) [reply src/hooks/useGas.ts:33]: yeah is the same but i removed it from there. oh wait is not removed, good catch i thought i did

[the remark this answers is not in the document]
Comment (estebanmino) [reply src/hooks/useGas.ts:34]: ok is removed now, i just left the methods here since is the only place we're using it

**Review by skylarbarrera — APPROVED** [review]

File: src/components/gas/GasSpeedButton.js [hunk 2]
```diff
@@ -173,9 +172,9 @@ const GasSpeedButton = ({
     () =>
       isNil(price) ||
       isEmpty(gasFeeParamsBySpeed) ||
-      isEmpty(selectedGasFee?.gasFee) ||
-      isSufficientGas === null,
-    [gasFeeParamsBySpeed, isSufficientGas, price, selectedGasFee]
+      isEmpty(selectedGasFee?.gasFee)[
+        (gasFeeParamsBySpeed, price, selectedGasFee)
+      ]
```
Comment (terrysahaidak) [inline src/components/gas/GasSpeedButton.js:177]: what the heck is going on here? :) 

I guess it's missing comma here
```suggestion
      isEmpty(selectedGasFee?.gasFee),
    [gasFeeParamsBySpeed, price, selectedGasFee]
```

**Review by terrysahaidak — CHANGES_REQUESTED** [review]
Seems like a typo

File: src/components/gas/GasSpeedButton.js (hunk 2 above)
Comment (estebanmino) [reply src/components/gas/GasSpeedButton.js:177]: wow thanks!

**Comment by estebanmino** [comment]
not sure  where did the `needs-qa` label went but this one needs qa as we talked @ibrahimtaveras00

**Review by osdnk — APPROVED** [review]

**Review by benisgold — APPROVED** [review]

**Review by ibrahimtaveras00 — APPROVED** [review]
Looks good on both OS's, QA Passed 👍🏽

**Pull request merged**